### PR TITLE
Update versions of python dependencies

### LINKED ISSUE #9699: MXNet pip package installs older version of numpy (1.13)
MXNet pip package installs numpy (<=1.13). We need to test and upgrade to the latest (1.14).
Some of the issues users can face:
1. Have an conda environment - with numpy, scipy etc. which is usually done by a DL practitioner.
2. Do pip install mxnet in the environment. This installs numpy 1.13.

Later user will fall into the following issue:

```
RuntimeError: module compiled against API version 0xc but this version of numpy is 0xb
ImportError: numpy.core.multiarray failed to import
ImportError: numpy.core.umath failed to import
ImportError: numpy.core.umath failed to import
```

due to conflicting versions of Numpy.

### PR DESCRIPTION
**Pull request by 0x1997**
## Description ##
Fix #9699. This should improve compatibility with other packages without introducing unexpected breakage.

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: python/setup.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
 else:
     from setuptools import setup
     from setuptools.extension import Extension
-    kwargs = {'install_requires': ['numpy<=1.13.3,>=1.8.2', 'requests==2.18.4', 'graphviz==0.8.1'], 'zip_safe': False}
+    kwargs = {'install_requires': ['numpy<1.15.0,>=1.8.2', 'requests<2.19.0,>=2.18.4', 'graphviz<0.9.0,>=0.8.1'], 'zip_safe': False}
```
Comment (marcoabreu) [inline python/setup.py]: Shouldn't we pin a specific version rather than specifying a range or is this the best practice in Python?

**Comment by marcoabreu** [comment]
Please also pin the same versions in https://github.com/apache/incubator-mxnet/blob/master/tests/ci_build/install/ubuntu_install_python.sh#L29 to provide a consistent behaviour. It seems like this problem slipped through because we are testing on the latest numpy version but expecting users to install a different one.

**Comment by marcoabreu** [comment]
Sheng, as far as I am aware, we're not using python/setup.py in CI.

File: python/setup.py (hunk 1 above)
Comment (szha) [reply python/setup.py]: Unlike setting up an environment where freezing all versions is preferable, we should accommodate the versions that users may have in the environment as much as we can. This change freezes the minor version, and we've tested all the minimum versions, so it's reasonable to expect newer versions not to break things.

**Comment by szha** [comment]
Should we use setup.py's dependencies?

**Comment by marcoabreu** [comment]
Yes, it would be good to have the same version in CI as in setup.py

**Comment by sandeep-krishnamurthy** [comment]
@marcoabreu - Does the updated changes from @0x1997  looks good? 
This PR fixes - https://github.com/apache/incubator-mxnet/issues/9888 and https://github.com/apache/incubator-mxnet/issues/9699

**Comment by marcoabreu** [comment]
LGTM. I have retriggered CI

**Comment by aaronmarkham** [comment]
@sandeep-krishnamurthy Our current build from source instructions for Mac tell you to use this script:
https://raw.githubusercontent.com/dmlc/mxnet/master/setup-utils/install-mxnet-osx-python.sh
Will it get the benefit of this PR, or does the script need updating too?

**Comment by CodingCat** [comment]
Hi, the community has passed to vote about associating the code changes with JIRA (https://lists.apache.org/thread.html/ab22cf0e35f1bce2c3bf3bec2bc5b85a9583a3fe7fd56ba1bbade55f@%3Cdev.mxnet.apache.org%3E)

We have updated the guidelines for contributors in https://cwiki.apache.org/confluence/display/MXNET/Development+Process, please ensure that you have created a JIRA at https://issues.apache.org/jira/projects/MXNET/issues/ to describe your work in this pull request and include the JIRA title in your PR as [MXNET-xxxx] your title where MXNET-xxxx is the JIRA id

Thanks!

**Comment by marcoabreu** [comment]
LGTM, but it has to be rebased and modified in order to match the new directory layout

**Comment by piiswrong** [comment]
looks like another PR fixed this. Closing

**Pull request closed without merging**